corehq/apps/hqadmin/reports.py

```diff
@@ -333,3 +336,84 @@ def _shorten_and_hyperlink_commit(self, commit_sha):
                 abbrev_sha=commit_sha[:7]
             )
         return None
+
+
+class UCRDataLoadReport(AdminReport):
+    UCR_RESTRICTION_THRESHOLD = 1_000_000
+
+    slug = 'ucr_data_load'
+    name = gettext_lazy("UCR Domains Data Report")
+
+    fields = [
+        'corehq.apps.reports.filters.simple.SimpleDomain',
+    ]
+    emailable = False
+    exportable = True
+    default_rows = 10
+
+    @property
+    def selected_domain(self):
+        selected_domain = self.request.GET.get('domain_name', None)
+        return selected_domain if selected_domain != '' else None
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn(gettext_lazy("Domain")),
+            DataTablesColumn(gettext_lazy("Case count")),
+            DataTablesColumn(gettext_lazy("Form count")),
+            DataTablesColumn(gettext_lazy("UCR rebuild restriction status")),
+        )
+
+    @property
+    def rows(self):
+        rows = []
+
+        for domain in self.ucr_domains():
+            rows.append(self._row_data(domain))
+
+        return rows
+
+    def ucr_domains(self):
+        ucr_domains = USER_CONFIGURABLE_REPORTS.get_enabled_domains()
+
+        if self.selected_domain:
+            if self.selected_domain in ucr_domains:
+                return [self.selected_domain]
+            return []
+
+        return ucr_domains
+
+    def _row_data(self, domain):
+        case_count = CaseES().domain(domain).count()
+        form_count = FormES().domain(domain).count()
+
+        return [
+            domain,
+            case_count,
+            form_count,
+            self.restrict_rebuild_column_data(domain, case_count, form_count),
+        ]
+
+    def restrict_rebuild_column_data(self, domain, case_count, form_count):
```

Comment:
nit: rename to `restrict_rebuild_column_data`

Reply:
Actually, now that I think about it, how does `_ucr_rebuild_restriction_status_column_data` sound? It's more in line with the column name.